HiSkim/HiOnia2MuMu/src/HiOnia2MuMuPAT.cc

```diff
@@ -520,31 +563,400 @@ HiOnia2MuMuPAT::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+
+    	  } else {
+    	    userBcFloat["vNChi2"] = -1;
+    	    userBcFloat["vProb"] = -1;
+    	    userBcFloat["vertexWeight"] = -100;
+    	    userBcFloat["sumPTPV"] = -100;
+    	    userBcFloat["ppdlPV"] = -100;
+    	    userBcFloat["ppdlErrPV"] = -100;
+    	    userBcFloat["cosAlpha"] = -100;
+    	    userBcFloat["ppdlBS"] = -100;
+    	    userBcFloat["ppdlErrBS"] = -100;
+    	    userBcFloat["ppdlOrigPV"] = -100;
+    	    userBcFloat["ppdlErrOrigPV"] = -100;
+    	    userBcFloat["ppdlPV3D"] = -100;
+    	    userBcFloat["ppdlErrPV3D"] = -100;
+    	    userBcFloat["cosAlpha3D"] = -100;
+    	    userBcFloat["ppdlBS3D"] = -100;
+    	    userBcFloat["ppdlErrBS3D"] = -100;
+    	    userBcFloat["ppdlOrigPV3D"] = -100;
+    	    userBcFloat["ppdlErrOrigPV3D"] = -100;
+
+    	    userBcInt["countTksOfPV"] = -1;
+
+    	    if (addCommonVertex_) {
+    	      userBcVertex["commonVertex"] = Vertex();
+    	    }
+    	    if (addMuonlessPrimaryVertex_) {
+    	      userBcVertex["muonlessPV"] = Vertex();
+    	      userBcVertex["PVwithmuons"] = Vertex();
+    	    } else {
+    	      userBcVertex["PVwithmuons"] = Vertex();
+    	    } 
+    	  }
+    	} else {
+    	  userBcFloat["vNChi2"] = -1;
+    	  userBcFloat["vProb"] = -1;
+    	  userBcFloat["vertexWeight"] = -100;
+    	  userBcFloat["sumPTPV"] = -100;
+    	  userBcFloat["ppdlPV"] = -100;
+    	  userBcFloat["ppdlErrPV"] = -100;
+    	  userBcFloat["cosAlpha"] = -100;
+    	  userBcFloat["ppdlBS"] = -100;
+    	  userBcFloat["ppdlErrBS"] = -100;
+    	  userBcFloat["ppdlOrigPV"] = -100;
+    	  userBcFloat["ppdlErrOrigPV"] = -100;
+    	  userBcFloat["ppdlPV3D"] = -100;
+    	  userBcFloat["ppdlErrPV3D"] = -100;
+    	  userBcFloat["cosAlpha3D"] = -100;
+    	  userBcFloat["ppdlBS3D"] = -100;
+    	  userBcFloat["ppdlErrBS3D"] = -100;
+    	  userBcFloat["ppdlOrigPV3D"] = -100;
+    	  userBcFloat["ppdlErrOrigPV3D"] = -100;
+        
+    	  userBcInt["countTksOfPV"] = -1;
+        
+    	  if (addCommonVertex_) {
+    	    userBcVertex["commonVertex"] = Vertex();
+    	  }
+    	  if (addMuonlessPrimaryVertex_) {
+    	    userBcVertex["muonlessPV"] = Vertex();
+    	    userBcVertex["PVwithmuons"] = Vertex();
+    	  } else {
+    	    userBcVertex["PVwithmuons"] = Vertex();
+    	  }
+    	}
+
+    	for (std::map<std::string, int>::iterator i = userBcInt.begin(); i != userBcInt.end(); i++) { BcCand.addUserInt(i->first , i->second); }
+    	for (std::map<std::string, float>::iterator i = userBcFloat.begin(); i != userBcFloat.end(); i++) { BcCand.addUserFloat(i->first , i->second); }
+    	for (std::map<std::string, reco::Vertex>::iterator i = userBcVertex.begin(); i != userBcVertex.end(); i++) { BcCand.addUserData(i->first , i->second); }
+
+	if(!LateTrimuonSel_(BcCand)) continue;	
+    	// ---- Push back output ----  
+    	trimuOutput->push_back(BcCand);
+
+      }//it3 muon
+    }//it2 muon
+  }//it muon
+
+  //  std::cout<<"number of selected Bc = "<<Bcnb<<std::endl;
```

Comment:
Please remove the print out lines that are commented

Reply:
Done